### ISSUE TITLE
java.lang.RuntimeException: Unknown exception code: 1 msg null

### ISSUE THREAD

**Issue by kassim** [hotchemi/PermissionsDispatcher#107]
Getting the following crash reported in an app I'm working on
```
Caused by java.lang.RuntimeException: Unknown exception code: 1 msg null
       at android.os.Parcel.readException(Parcel.java:1484)
       at android.os.Parcel.readException(Parcel.java:1426)
       at android.app.ActivityManagerProxy.checkPermission(ActivityManagerNative.java:3450)
       at android.app.ContextImpl.checkPermission(ContextImpl.java:1703)
       at android.content.ContextWrapper.checkPermission(ContextWrapper.java:570)
       at android.content.ContextWrapper.checkPermission(ContextWrapper.java:570)
       at android.support.v4.content.PermissionChecker.checkPermission(PermissionChecker.java:89)
       at android.support.v4.content.PermissionChecker.checkSelfPermission(PermissionChecker.java:125)
       at permissions.dispatcher.PermissionUtils.hasSelfPermissions(PermissionUtils.java:40)
```

I'm not sure what the cause is but looking at this issue reported in the Appboy SDK https://github.com/Appboy/appboy-android-sdk/issues/42 their solution was to wrap the call in a try/catch

Do you know what may be the cause?

**Comment by shiraji** [comment]
Could you give us a minimum project that reproduces this issue?

**Comment by aurae** [comment]
Are you experiencing this consistently on a particular device? What's the OS version on it? This issue is most definitely related to the support library's `PermissionChecker` class. We'll have to check the Android source in order to find the root of this. As you said, both Appboy and PayPal wrap their `checkSelfPermission` invocation in a try/catch and that's it, which seems to be fine, however I'd like to find out what's causing this first.

**Comment by kassim** [comment]
I can't reproduce it locally unfortunately. 
Lenovo A816 on 4.4.4 is the main device we're seeing it reported on, when retrieving Contacts permissions in my case.

**Comment by aurae** [comment]
This is the offending implementation of `Parcel#readException`:

```java
/**
 * Use this function for customized exception handling.
 * customized method call this method for all unknown case
 * @param code exception code
 * @param msg exception message
 */
public final void readException(int code, String msg) {
    switch (code) {
        case EX_SECURITY:
            throw new SecurityException(msg);
        case EX_BAD_PARCELABLE:
            throw new BadParcelableException(msg);
        case EX_ILLEGAL_ARGUMENT:
            throw new IllegalArgumentException(msg);
        case EX_NULL_POINTER:
            throw new NullPointerException(msg);
        case EX_ILLEGAL_STATE:
            throw new IllegalStateException(msg);
        case EX_NETWORK_MAIN_THREAD:
            throw new NetworkOnMainThreadException();
    }
    throw new RuntimeException("Unknown exception code: " + code
            + " msg " + msg);
}
```

Since the default `EX_` constants are all negative and no suitable `case` exists for code 1, the `RuntimeException` is raised. I haven't found the meaning of this status code yet. For now I think that we should simply adjust to what everybody else does, assume the non-zero result to be an error trying to obtain the Permission state, and return false in `PermissionUtils`.

**Comment by hotchemi** [comment]
Closed by #108 .
@kassim Do you need new version soon?

**Comment by hotchemi** [comment]
@kassim Just released 2.0.8 to jcenter.

**Comment by kassim** [comment]
Cheers guys, much appreciated.
